pkg/lock/lock_nix.go

```diff
@@ -0,0 +1,70 @@
+// +build !windows,!plan9
+
+package lock
+
+import (
+	"os"
+	"syscall"
+)
+
+var (
+	wrlck = syscall.Flock_t{
+		Type:   syscall.F_WRLCK,
+		Whence: 0,
+		Start:  0,
+		Len:    0,
+		Pid:    0,
+	}
+	rdlck = syscall.Flock_t{
+		Type:   syscall.F_RDLCK,
+		Whence: 0,
+		Start:  0,
+		Len:    0,
+		Pid:    0,
+	}
+	unlck = syscall.Flock_t{
+		Type:   syscall.F_UNLCK,
+		Whence: 0,
+		Start:  0,
+		Len:    0,
+		Pid:    0,
+	}
+)
+
+func open(path string, flag int, perm os.FileMode) (*os.File, error) {
+	return os.OpenFile(path, flag, perm)
+}
+
+func lock(f *os.File) error {
+	lock := wrlck
+	lock.Pid = int32(os.Getpid())
```

Comment:
The pid of minio server process will not change. You can initialize the Pid field at the time of initializing ``wrlck`` and ``rdlck`` variables.